src/common/utils.ts

```diff
@@ -83,6 +84,34 @@ export const generateGeoIdentifier = (geo: FeatureCollection): string => {
   return additionalIdentifiers;
 };
 
+export function createSpanMetadata(
+  functioName?: string,
+  spanKind?: SpanKind,
+  context?: { traceId: string; spanId: string }
+): { traceContext: SpanContext | undefined; spanOptions: SpanOptions | undefined } {
+  const FLAG_SAMPLED = 1;
+  if (!context) {
+    return { spanOptions: undefined, traceContext: undefined };
+  }
+  const traceContext: SpanContext = {
```

Comment:
check if we can work with this type every time we use your custom parent context interface